include/git2/notes.h

```diff
@@ -153,6 +153,33 @@ GIT_EXTERN(int) git_note_create(
 	const char *note,
 	int force);
 
+/**
+ * Add a note for an object from a commit
+ *
+ * @param notes_commit_out pointer to store the commit (optional);
+ *					NULL in case of error
+ * @param notes_blob_out a point to the id of a note blob (optional)
+ * @param repo repository where the note will live
+ * @param notes_commit a pointer to the notes commit object
+ *					or NULL if this is to be the first notes object
+ * @param author signature of the notes commit author
+ * @param committer signature of the notes commit committer
+ * @param oid OID of the git object to decorate
+ * @param note Content of the note to add for object oid
+ * @param force Overwrite existing note
+ *
+ * @return 0 or an error code
+ */
+GIT_EXTERN(int) git_note_commit_create(
+	git_commit **notes_commit_out,
+	git_oid *notes_blob_out,
+	git_repository *repo,
+	git_commit *notes_commit,
+	const git_signature *author,
+	const git_signature *committer,
+	const git_oid *oid,
+	const char *note,
+	int allow_note_overwrite);
```

Comment:
I'm currently trying to understand this function. As I understand, a note is created for an object (e.g. either a blob, tree, commit or tag) inside of a specific reference. At least, this is how I understand `git notes add`.

So this function here removes the note reference parameter from `git_note_create` which hints where the note should be put in, and instead uses an existing notes commit as its parent. And subsequently, we do not store the resulting notes commit in any reference, instead it's created as a dangling commit.

So this buys us the ability to easily create a complete notes tree without having to write and re-read the notes reference, correct? And in the end, the developer is free to create the notes reference for himself.